network/fabrics_test.go

```diff
@@ -0,0 +1,875 @@
+package network_test
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+
+	"strconv"
+
+	"io/ioutil"
+
+	"github.com/joyent/triton-go/network"
+	"github.com/joyent/triton-go/testutils"
+	"github.com/pkg/errors"
+)
+
+var (
+	fakeFabricVlanId      = 2
+	deleteFabricErrorType = errors.New("unable to delete fabric")
+	deleteVLanErrorType   = errors.New("unable to delete vlan")
+	createVLanErrorType   = errors.New("unable to create vlan")
+	createFabricErrorType = errors.New("unable to create fabric")
+	getVlanErrorType      = errors.New("unable to get vlan")
+	getFabricErrorType    = errors.New("unable to get fabric")
+	updateVlanErrorType   = errors.New("unable to update vlan")
+	listVlansErrorType    = errors.New("unable to list vlans")
+	listFabricsErrorType  = errors.New("unable to list fabrics")
+)
+
+func TestDeleteFabric(t *testing.T) {
+	networkClient := MockNetworkClient()
+
+	do := func(ctx context.Context, nc *network.NetworkClient) error {
+		defer testutils.DeactivateClient()
+
+		return nc.Fabrics().Delete(ctx, &network.DeleteFabricInput{
+			FabricVLANID: fakeFabricVlanId,
+			NetworkID:    fakeNetworkId,
+		})
+	}
+
+	t.Run("successful", func(t *testing.T) {
+		testutils.RegisterResponder("DELETE", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks", fakeNetworkId), deleteFabricSuccess)
```

Comment:
Can we move these registrations to above the `t.Run()` call so that the mock responder has all routes loaded?  Then the calls to `Run()` are acting on a static input and could, in theory, be run concurrently as parallel tests.

I see the appeal of putting the `RegisterResponder` in the test `Run()` itself, but think it's better to comment each `RegisterResponder` once so all of the `Run()` operations have stable input and we're not potentially adding to the mock handlers and stomping on old behavior in the same test case.